src/dos/drives.cpp

```diff
@@ -25,20 +25,32 @@
 
 extern char sfn[DOS_NAMELENGTH_ASCII];
 
+constexpr bool is_special_character(const char c)
+{
+	constexpr auto special_characters = std::string_view("\"+=,;:<>[]|?*");
+	return special_characters.find(c) != std::string_view::npos;
+}
+
 // TODO Right now label formatting seems to be a bit of mess, with various
 // places in code setting/expecting different format, so simple GetLabel() on
 // a drive object might not yield an expected result. Not sure how to sort it
 // out, but it will require some attention to detail.
-// Also: this function is too strict - it removes all punctuation when *some*
-// punctuation is acceptable in drive labels (e.g. '_' or '-').
 //
 std::string To_Label(const char* name) {
 	// Reformat the name per the DOS label specification:
 	// - Upper-case, up to 11 ASCII characters
 	// - Internal spaces allowed but no: tabs ? / \ | . , ; : + = [ ] < > " '
 	std::string label(name);
 	trim(label); // strip front-and-back white-space
-	strip_punctuation(label); // strip all punctuation
+	label.erase(std::remove_if(label.begin(),
```

Comment:
Do you need this outer `label.erase()`?  Doesn't `std::remove_if()` handle everything?  From the docs here:  https://en.cppreference.com/w/cpp/algorithm/remove

What I'm reading is at best the `label.erase()` is a no-op because `std::remove_if()` returns an end iterator.  At worst, it might be undefined behavior because the compiler could evaluate `label.end()` before `std::remove_if()` and then the end iterator becomes invalidated.